In [49]:
import pandas as pd
import numpy as np

Loding the data file.

In [50]:
fundingdata=pd.read_csv('userssharedsdfschoolimprovement2010grants.csv')

In [51]:
fundingdata.head()

School Name      City State  \
0  HOGARTH KINGEEKUK MEMORIAL SCHOOL  SAVOONGA    AK   
1                    AKIACHAK SCHOOL  AKIACHAK    AK   
2                     GAMBELL SCHOOL   GAMBELL    AK   
3               BURCHELL HIGH SCHOOL   WASILLA    AK   
4                       AKIAK SCHOOL     AKIAK    AK   

                               District Name 2010/11/Award Amount  \
0              BERING STRAIT SCHOOL DISTRICT           $471014.00   
1                     YUPIIT SCHOOL DISTRICT           $520579.00   
2              BERING STRAIT SCHOOL DISTRICT           $449592.00   
3  MATANUSKA-SUSITNA BOROUGH SCHOOL DISTRICT           $641184.00   
4                     YUPIIT SCHOOL DISTRICT           $399686.00   

   Model Selected                                           Location  
0  Transformation  200 MAIN ST\nSAVOONGA, AK 99769\n(63.6687, -17...  
1  Transformation  AKIACHAK 51100\nAKIACHAK, AK 99551\n(60.8911, ...  
2  Transformation  169 MAIN ST\nGAMBELL, AK 99742\n(63.7413, -171...  
3  Transformation  1775 WEST PARKS HWY\nWASILLA, AK 99654\n(61.57...  
4  Transformation     AKIAK 5227\nAKIAK, AK 99552\n(60.8879, -161.2)

In [52]:
fundingdata.tail()

School Name        City State  \
826         MOUNT HOPE HIGH SCHOOL  MOUNT HOPE    WV   
827        GEARY ELEMENTARY SCHOOL   LEFT HAND    WV   
828            TRIUMPH HIGH SCHOOL    CHEYENNE    WY   
829  HEM JUNIOR/SENIOR HIGH SCHOOL       HANNA    WY   
830          ROOSEVELT HIGH SCHOOL      CASPER    WY   

                         District Name 2010/11/Award Amount  Model Selected  \
826             FAYETTE COUNTY SCHOOLS           $342189.00  Transformation   
827               ROANE COUNTY SCHOOLS           $260000.00  Transformation   
828  LARAMIE COUNTY SCHOOL DISTRICT #1           $297300.00  Transformation   
829   CARBON COUNTY SCHOOL DISTRICT #2           $274000.00  Transformation   
830  NATRONA COUNTY SCHOOL DISTRICT #1           $431382.00  Transformation   

                                              Location  
826  110 HIGH SCHOOL DR\nMOUNT HOPE, WV 25880\n(37....  
827  9538 CLAY RD\nLEFT HAND, WV 25251\n(38.6188, -...  
828  2201 MORRIE AVE\nCHEYENNE, WY 82001\n(41.1423,...  
829    US HWY 72\nHANNA, WY 82327\n(41.8723, -106.548)  
830  140 EAST K ST\nCASPER, WY 82601\n(42.861, -106...

Pandas is showing 831 records. Once Philadelphia schools data is separated out, check if 27 schools are shown.Check using describe.

In [53]:
fundingdata.describe()

School Name          City State         District Name  \
count            831           831   831                   831   
unique           821           424    50                   421   
top     CENTRAL HIGH  PHILADELPHIA    CA  PHILADELPHIA CITY SD   
freq               2            29    92                    27   

       2010/11/Award Amount  Model Selected  \
count                   757             825   
unique                  651               4   
top                   $2.00  Transformation   
freq                     17             608   

                                                 Location  
count                                                 831  
unique                                                823  
top     6941 NORTH CENTRAL ST\nPORTLAND, OR 97203\n(45...  
freq                                                    3

The count is 831. Index stops at 830 becasue Python reads from 0.

Separate out only Phila School District Items.

In [54]:
philafunds=fundingdata[fundingdata['District Name']. isin(["PHILADELPHIA CITY SD"])]   
philafunds.head()

School Name          City State         District Name  \
603     SAYRE WILLIAM L MS  PHILADELPHIA    PA  PHILADELPHIA CITY SD   
605  KING MARTIN LUTHER HS  PHILADELPHIA    PA  PHILADELPHIA CITY SD   
606  DOUGLAS STEPHEN A SCH  PHILADELPHIA    PA  PHILADELPHIA CITY SD   
609        LOCKE ALAIN SCH  PHILADELPHIA    PA  PHILADELPHIA CITY SD   
610       ALCORN JAMES SCH  PHILADELPHIA    PA  PHILADELPHIA CITY SD   

    2010/11/Award Amount  Model Selected  \
603           $737032.00  Transformation   
605           $786964.00  Transformation   
606           $319320.00         Restart   
609           $390938.00  Transformation   
610           $338726.00  Transformation   

                                              Location  
603  5800 WALNUT ST\nPHILADELPHIA, PA 19139\n(39.95...  
605  6100 STENTON AVE\nPHILADELPHIA, PA 19138\n(40....  
606  2700 EAST HUNTINGDON ST\nPHILADELPHIA, PA 1912...  
609  4550 HAVERFORD AVE\nPHILADELPHIA, PA 19139\n(3...  
610  3200 DICKINSON ST\nPHILADELPHIA, PA 19146\n(39...

In [55]:
philafunds.describe()

School Name          City State         District Name  \
count              27            27    27                    27   
unique             27             1     1                     1   
top     ROXBOROUGH HS  PHILADELPHIA    PA  PHILADELPHIA CITY SD   
freq                1            27    27                    27   

       2010/11/Award Amount  Model Selected  \
count                    27              27   
unique                   27               3   
top              $244855.00  Transformation   
freq                      1              16   

                                                 Location  
count                                                  27  
unique                                                 27  
top     3601 FILBEROUTE ST\nPHILADELPHIA, PA 19104\n(3...  
freq                                                    1

The describe() method is currently not yeilding the statistics. This is likely becasue the Award Amounts are not integers or floats. Check Data Type.

In [56]:
philafunds.dtypes

School Name             object
City                    object
State                   object
District Name           object
2010/11/Award Amount    object
Model Selected          object
Location                object
dtype: object

Need to convert Award Amount Column to INtegers or Floats. the only non-numerical charecter in Amounts is $. By removing the $ sign it should allow defining floats. 

In [57]:
philafunds['2010/11/Award Amount']= philafunds['2010/11/Award Amount'].str.replace('$','')
philafunds.head()

C:\Users\Veeru\Anaconda3\lib\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.


School Name          City State         District Name  \
603     SAYRE WILLIAM L MS  PHILADELPHIA    PA  PHILADELPHIA CITY SD   
605  KING MARTIN LUTHER HS  PHILADELPHIA    PA  PHILADELPHIA CITY SD   
606  DOUGLAS STEPHEN A SCH  PHILADELPHIA    PA  PHILADELPHIA CITY SD   
609        LOCKE ALAIN SCH  PHILADELPHIA    PA  PHILADELPHIA CITY SD   
610       ALCORN JAMES SCH  PHILADELPHIA    PA  PHILADELPHIA CITY SD   

    2010/11/Award Amount  Model Selected  \
603            737032.00  Transformation   
605            786964.00  Transformation   
606            319320.00         Restart   
609            390938.00  Transformation   
610            338726.00  Transformation   

                                              Location  
603  5800 WALNUT ST\nPHILADELPHIA, PA 19139\n(39.95...  
605  6100 STENTON AVE\nPHILADELPHIA, PA 19138\n(40....  
606  2700 EAST HUNTINGDON ST\nPHILADELPHIA, PA 1912...  
609  4550 HAVERFORD AVE\nPHILADELPHIA, PA 19139\n(3...  
610  3200 DICKINSON ST\nPHILADELPHIA, PA 19146\n(39...

In [58]:
philafunds['2010/11/Award Amount']= philafunds['2010/11/Award Amount'].astype(float)
philafunds.head()


C:\Users\Veeru\Anaconda3\lib\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.


School Name          City State         District Name  \
603     SAYRE WILLIAM L MS  PHILADELPHIA    PA  PHILADELPHIA CITY SD   
605  KING MARTIN LUTHER HS  PHILADELPHIA    PA  PHILADELPHIA CITY SD   
606  DOUGLAS STEPHEN A SCH  PHILADELPHIA    PA  PHILADELPHIA CITY SD   
609        LOCKE ALAIN SCH  PHILADELPHIA    PA  PHILADELPHIA CITY SD   
610       ALCORN JAMES SCH  PHILADELPHIA    PA  PHILADELPHIA CITY SD   

     2010/11/Award Amount  Model Selected  \
603              737032.0  Transformation   
605              786964.0  Transformation   
606              319320.0         Restart   
609              390938.0  Transformation   
610              338726.0  Transformation   

                                              Location  
603  5800 WALNUT ST\nPHILADELPHIA, PA 19139\n(39.95...  
605  6100 STENTON AVE\nPHILADELPHIA, PA 19138\n(40....  
606  2700 EAST HUNTINGDON ST\nPHILADELPHIA, PA 1912...  
609  4550 HAVERFORD AVE\nPHILADELPHIA, PA 19139\n(3...  
610  3200 DICKINSON ST\nPHILADELPHIA, PA 19146\n(39...

In [59]:
philafunds.describe()

2010/11/Award Amount
count             27.000000
mean          555936.518519
std           288151.681054
min                1.000000
25%           336026.500000
50%           579958.000000
75%           793011.000000
max           980283.000000

The Amounts objects are now float objects. The describe()function above shows the statistics - including mean and Std Deviation.